### ISSUE TITLE
[SL/SL2] Speakers hissing while booting

### ISSUE THREAD

**Issue by SoleSensei** [jakeday/linux-surface#354]
Laptop surface. Lastest kernel release.
A static hiss from my speakers a few seconds after grub2 screen (dual booting)
#255 #101

**Comment by SoleSensei** [comment]
and when i closing laptop cover - laptop is switching off with the same hiss sound
i think it is hibernate problem

**Comment by rwx777** [comment]
My surface laptop has the same issue. And it keeps hissing from SL speakers when switching audio output to the HDMI. SL1/ 941c0e8/ kernel 4.18.20.

**Comment by SoleSensei** [comment]
> when lid is closed - laptop is shut-downing

### Some tweaks to fix (Ubuntu 18.04):
#### Disable suspend on lid close
1. Open the `/etc/systemd/logind.conf` file in a text editor as root, for example:
```bash
sudo -H nano /etc/systemd/logind.conf
```
2. Add lines (or change if uncommented) `HandleLidSwitch=ignore` and 'HandleLidSwitchDocked=ignore'
3. Restart service
```bash
sudo service systemd-logind restart
```
#### Disable screen on lid close
1. Run
```bash
sudo -i
echo 'event=button/lid.*' | tee --append /etc/acpi/events/lm_lid
echo 'action=/etc/acpi/lid.sh' | tee --append /etc/acpi/events/lm_lid
touch /etc/acpi/lid.sh
chmod +x /etc/acpi/lid.sh
```
2. Open `/etc/acpi/lid.sh` in text editor
```bash
sudo -H nano /etc/acpi/lid.sh
```
3. Copy-Paste
```bash
#!/bin/bash
 
USER=your_username
 
grep -q close /proc/acpi/button/lid/*/state
 
if [ $? = 0 ]; then
  su -c  "sleep 1 && xset -display :0.0 dpms force off" - $USER
fi
```
4. Replace `your_username` with your name

**Comment by btearle** [comment]
I have the same issue. It goes away after the desktop loads, but I just realized it comes back whenever I plug headphones in (the hissing still coming from the computer speakers, not the headphones.)

**Comment by dotvezz** [comment]
I'm getting the same behavior now. It had been okay before the acpi module, but now the static/hiss noise starts up every time I change the battery charging state (plug/unplug the charger) and every time I connect headphones.

The amixer command on the [Archwiki page](https://wiki.archlinux.org/index.php/Microsoft_Surface_Book_2#Audio) had worked for me before, but doesn't seem to do anything now.

**Comment by rwx777** [comment]
@SoleSensei 
Yes, it shutdown directly. I need to press power again to open it. However, I have not tried the tweak yet.

**Comment by paulhutchings** [comment]
Disabling the on-board audio via UEFI is the only way to get rid of it (of course, then you have no audio though). My guess is it's either activating the mic and boosting the gain like crazy resulting in the feedback, or it's just booting the gain on the speakers like crazy and what you're hearing is the noise floor.

**Comment by mace07** [comment]
Confirming same issue with SL1, newest kernel. Hiss will occur during boot for a few seconds and end with a pop. Will intermittently occur again when booted and logged in. I've found that just starting to play an mp3 file will produce the pop and stop the hissing, until it occurs again.

**Comment by jakeday** [comment]
Can someone please test this on 4.19.23-1?

**Comment by gloveboxes** [comment]
Hey there , hiss on startup and login on Surface Laptop 2 still there...

**Comment by rwx777** [comment]
Surface Laptop 1, just install 4.19.23-1, still hissing. Should I give the log file or something? maybe it will help.

**Comment by SoleSensei** [comment]
> Can someone please test this on 4.19.23-1?
@jakeday 
Tested. Surface Laptop. Hissing on boot.

**Comment by stefanpoensgen** [comment]
i don't think this is related to the linux surface kernel. i even noticed this with the 4.15 kernel in ubuntu 18.04.

**Comment by ezyang** [comment]
Also observing hissing on Surface Book 2 at boot.

**Comment by SoleSensei** [comment]
I also found this issue while booting via live usb, but that hissing was much louder than surface-kernel

**Comment by remingtonlang** [comment]
`amixer -c 0 sset 'Auto-Mute Mode' Disabled && sudo alsactl store` just fyi for everyone will fix any hissing or pops after the device boots. @paulhutchings is correct. the issue is the feedback from the MIC

**Comment by dotvezz** [comment]
Hi @remingtonlang,

I'd used that before on a previous kernel, but IIRC it hasn't worked in a while. Will try and confirm when I get home though.

**Comment by stefanpoensgen** [comment]
> `amixer -c 0 sset 'Auto-Mute Mode' Disabled && sudo alsactl store` just fyi for everyone will fix any hissing or pops after the device boots. @paulhutchings is correct. the issue is the feedback from the MIC

still hissing on my sb2 with ubuntu 18.04

**Comment by dotvezz** [comment]
Confirming @stefanpoensgen's findings on my 15" SB2 on Arch Linux. The command @remingtonlang shared doesn't change anything.

That tip did work on previous Surface kernels, but hasn't been working since ... I want to say 4.18.xx? I think about the same time we got ACPI modules, it stopped working.

**Comment by remingtonlang** [comment]
Create this file and add the following

`$ cat /etc/modprobe.d/sound.conf 
options snd-hda-intel model=generic
`

reboot, any difference? I'm not that old yet that I can't hear it but it makes a notable difference AFAIK

**Comment by stefanpoensgen** [comment]
> Create this file and add the following
> 
> > $ cat /etc/modprobe.d/sound.conf
> > options snd-hda-intel model=generic
> 
> reboot, any difference? I'm not that old yet that I can't hear it but it makes a notable difference AFAIK

the hissing is gone. There are only two short noises, but very very quiet. ok for me

**Comment by dbranham20** [comment]
> Create this file and add the following
> 
> > $ cat /etc/modprobe.d/sound.conf
> > options snd-hda-intel model=generic
> 
> reboot, any difference? I'm not that old yet that I can't hear it but it makes a notable difference AFAIK

Tried this on Surface Laptop 2. Didn't have a `sound.conf` file so I made one. Added that line and the laptop no longer hisses or pops on login

**Comment by Willcher** [comment]
> Create this file and add the following
> 
> > $ cat /etc/modprobe.d/sound.conf
> > options snd-hda-intel model=generic
> 
> reboot, any difference? I'm not that old yet that I can't hear it but it makes a notable difference AFAIK

On Surface Book 2 there's no hissing, but now I only have audio coming through the headphone jack (before only had audio coming through the speakers, and only hissing through the main speakers with headphones plugged in).

**Comment by remingtonlang** [comment]
@Willcher for hissing with headphones in my previous amixer commands address that.

**Comment by Willcher** [comment]
> @Willcher for hissing with headphones in my previous amixer commands address that.

Okay that command still works for me. For some reason wasn't getting any audio at all through the headphone jack regardless of that setting before but that issue went away after rebooting.

**Comment by SoleSensei** [comment]
for me all hissing sounds are gone somehow.
```
Linux Surface 4.19.27-surface-linux-surface
```
```
amixer -c 0 sset 'Auto-Mute Mode' Disabled && sudo alsactl store
```

**Comment by SoleSensei** [comment]
Retest on `Linux Surface 5.0.1`, no hissing

**Comment by xjcl** [comment]
Tested on SL1, I can confirm @stefanpoensgen's observations (2 short quiet noises instead of hissing+pop)

**Comment by runekaagaard** [comment]
Tested on Ubuntu 18.04.3, qzed's release at https://github.com/jakeday/linux-surface/pull/540. No hissing during install or boot. Headphones causes speakers to hiss. The command `amixer -c 0 sset 'Auto-Mute Mode' Disabled && sudo alsactl store` fixes that problem and persists during boot.
